### PR TITLE
Add `JobRun` state field

### PR DESCRIPTION
**Pull request by dalbhanj**
Issue https://github.com/aws-controllers-k8s/community/issues/1489

Description of changes: Kubectl can show state changes of jobruns

- Add `Status.State` field
- Add kubebuilder columns to support printing State and ID when calling `kubectl get`

Example output:

```
$ kubectl get jobruns -w
NAME               JOB-ID                STATE
my-ack-jobrun-01   0000000312np262is2t   FAILED
my-ack-jobrun-02   0000000312nrulbojo3   COMPLETED
my-ack-jobrun-03   0000000312t1q0qonkc   COMPLETED
my-ack-jobrun-04   0000000312uesu2s8cu   COMPLETED
my-ack-jobrun-05   0000000312uhf8rs910   COMPLETED
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by A-Hilaly — APPROVED** [review]
LGTM
+++

File: pkg/resource/job_run/delta.go [hunk 1]
```diff
@@ -105,6 +105,13 @@ func newResourceDelta(
 	if !reflect.DeepEqual(a.ko.Spec.VirtualClusterRef, b.ko.Spec.VirtualClusterRef) {
 		delta.Add("Spec.VirtualClusterRef", a.ko.Spec.VirtualClusterRef, b.ko.Spec.VirtualClusterRef)
 	}
+	if ackcompare.HasNilDifference(a.ko.Spec.ConfigurationOverrides, b.ko.Spec.ConfigurationOverrides) {
+		delta.Add("Spec.ConfigurationOverrides", a.ko.Spec.ConfigurationOverrides, b.ko.Spec.ConfigurationOverrides)
+	} else if a.ko.Spec.ConfigurationOverrides != nil && b.ko.Spec.ConfigurationOverrides != nil {
+		if *a.ko.Spec.ConfigurationOverrides != *b.ko.Spec.ConfigurationOverrides {
+			delta.Add("Spec.ConfigurationOverrides", a.ko.Spec.ConfigurationOverrides, b.ko.Spec.ConfigurationOverrides)
+		}
+	}
```
Comment (A-Hilaly) [inline pkg/resource/job_run/delta.go:114]: Need to investigate why is this generated again
/hold

Comment (dalbhanj) [reply pkg/resource/job_run/delta.go]: Looks like there is a bug in code-generator latest commit. Reverting to previous commit (2944c87) solves the issue

**Comment by A-Hilaly** [comment]
@dalbhanj can you try regenerating with code-generator 5ee0ac052c54f008dff50f6f5ebb73f2cf3a0bd7 instead? looks like we lost few helm features in this PR.

**Comment by dalbhanj** [comment]
> @dalbhanj can you try regenerating with code-generator 5ee0ac052c54f008dff50f6f5ebb73f2cf3a0bd7 instead? looks like we lost few helm features in this PR.

Done

File: apis/v1alpha1/generator.yaml [hunk 2]
```diff
@@ -45,13 +45,27 @@ resources:
...
+          name: STATE
+        from:
+          operation: DescribeJobRun
+          path: JobRun.State       
+      Id:
+        is_read_only: true
+        print:
+          name: Job-ID
       # The default ConfigurationOverrides that is part of JobRun API's does not work well with ACK. By design a CRD must be describable using an OpenAPIv3 schema. ConfigurationOverrides has a property within ApplicationConfiguration called 'configurations' which is recursive and is the root of the issue. Hence, we are creating custom JSON blob to accomodate ConfigurationOverrides and we have to manually handle how to unmarshall this field so that customers can define all/any properties available
       configurationOverrides:
```
Comment (A-Hilaly) [inline apis/v1alpha1/generator.yaml:60]: This should be "ConfigurationOverrides" that's why code generation didn't work properly earlier

File: apis/v1alpha1/generator.yaml [hunk 3]
```diff
@@ -45,20 +45,34 @@ resources:
...
+    reconcile: 
+      requeue_on_success_seconds: 15
     hooks:
       delta_pre_compare:
         code: customPreCompare(delta, a, b)
       sdk_create_post_build_request:
-        template_path: hooks/configurationOverrides/sdk_create_post_build_request.go.tpl
+        template_path: hooks/ConfigurationOverrides/sdk_create_post_build_request.go.tpl
       sdk_read_one_pre_set_output:
-        template_path: hooks/configurationOverrides/sdk_read_one_pre_set_output.go.tpl
```
Comment (A-Hilaly) [inline apis/v1alpha1/generator.yaml:61]: Please keep this `configurationOverrides` because that's the directory of the hooks

**Review by A-Hilaly — CHANGES_REQUESTED** [review]

File: apis/v1alpha1/generator.yaml (hunk 3 above)
Comment (dalbhanj) [reply apis/v1alpha1/generator.yaml:61]: changing path to configuration_overrides to keep it compatible with other repos.

File: apis/v1alpha1/generator.yaml (hunk 2 above)
Comment (dalbhanj) [reply apis/v1alpha1/generator.yaml]: Done

**Review by A-Hilaly — APPROVED** [review]
/lgtm

**Review by A-Hilaly — APPROVED** [review]
/lgtm
/unhold

**Pull request merged**